docs/connecting-navigation-prop.md

```diff
@@ -4,6 +4,7 @@ title: Access the navigation prop from any component
 sidebar_label: Access the navigation prop from any component
 ---
 
+#TODO maybe we should support it?
```

Comment:
Change the docs to use the `useNavigation` hook since that's what we have now.
We can add the HOC version to the compat lib